task/backend/coordinator/task_coordinator.go

```diff
@@ -0,0 +1,183 @@
+package coordinator
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/influxdata/influxdb"
+	"github.com/influxdata/influxdb/task/backend"
+	"github.com/influxdata/influxdb/task/backend/executor"
+	"github.com/influxdata/influxdb/task/backend/middleware"
+	"github.com/influxdata/influxdb/task/backend/scheduler"
+	"go.uber.org/zap"
+)
+
+var _ middleware.Coordinator = (*Coordinator)(nil)
+var _ Executor = (*executor.TaskExecutor)(nil)
+var _ scheduler.SchedulableService = (*SchedulableTaskService)(nil)
+
+// Executor is an abstraction of the task executor with only the functions needed by the coordinator
+type Executor interface {
+	ManualRun(ctx context.Context, id scheduler.ID, runID scheduler.ID) (executor.Promise, error)
```

Comment:
These should use `influxdb.ID`. Why would a communication between the coordinator and the executor need to communicate using an ID thats non standard for the influxdb system? The scheduler only uses its own id because we want it to be able to stand on its own.

Reply:
good point, I'll fix that